packages/flutter/lib/src/rendering/object.dart

```diff
@@ -3248,30 +3249,87 @@ abstract class RenderObject with DiagnosticableTreeMixin implements HitTestTarge
   /// the global coordinate system in logical pixels. To get physical pixels,
   /// use [applyPaintTransform] from the [RenderView] to further transform the
   /// coordinate.
+  ///
+  /// See also:
+  ///
+  ///  * [computeTransformToRenderObject] which does the same, but the target
+  ///    `RenderObject` does not have to be an ancestor.
   Matrix4 getTransformTo(RenderObject? ancestor) {
```

Comment:
For my own understanding: Why refactor this method as well? What is the benefit? Is it faster? (At first I thought it was to make code sharable between the two methods, but that doesn't seem to be the case?)

FWIW, for my personal taste, the old implementation was more self-documenting and easier to follow along what it does. It was also only half as long. The new implementation is less straight-forward and introduces separate code paths depending on the argument...